### PR TITLE
Add Support for text/plain to Azure Core

### PR DESCRIPTION
**Pull request by alzimmermsft**
This PR adds support for handling sending and receiving `text/plain` `Content-Type` requests and responses.

### CODE REVIEW

File: sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/JacksonAdapter.java [hunk 1]
```diff
@@ -170,6 +171,51 @@ public String serializeList(List<?> list, CollectionFormat format) {
...
+        } else if (type.isAssignableFrom(double.class)) {
+            convertedValue = Double.parseDouble(value);
+        } else if (type.isAssignableFrom(Double.class)) {
+            convertedValue = Double.valueOf(value);
+        } else if (type.isAssignableFrom(boolean.class)) {
+            convertedValue = Boolean.parseBoolean(value);
+        } else if (type.isAssignableFrom(Boolean.class)) {
+            convertedValue = Boolean.valueOf(value);
+        } else {
+            // Should this check if there are any String only constructors or static factories?
```
Comment (jianghaolu) [inline sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/JacksonAdapter.java:212]: like enums and expandable enums?

Comment (alzimmermsft) [reply sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/JacksonAdapter.java:212]: Yeah that is one scenario

Comment (jianghaolu) [reply sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/JacksonAdapter.java:212]: I think supporting ExpandableStringEnum should be pretty straightforward. For others we may have to reflect on the class to get the constructor/ static method signatures, which I'm not sure if that's worth it.

File: sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/JacksonAdapter.java [hunk 2]
```diff
@@ -170,6 +171,51 @@ public String serializeList(List<?> list, CollectionFormat format) {
...
+            convertedValue = Float.valueOf(value);
+        } else if (type.isAssignableFrom(double.class)) {
+            convertedValue = Double.parseDouble(value);
+        } else if (type.isAssignableFrom(Double.class)) {
+            convertedValue = Double.valueOf(value);
+        } else if (type.isAssignableFrom(boolean.class)) {
+            convertedValue = Boolean.parseBoolean(value);
+        } else if (type.isAssignableFrom(Boolean.class)) {
+            convertedValue = Boolean.valueOf(value);
+        } else {
```
Comment (alzimmermsft) [inline sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/JacksonAdapter.java:211]: Add checks for `enum` and `ExpandableStringEnum`.

**Comment by JonathanGiles** [comment]
@alzimmermsft Are you going to work on getting this updated and merged?

**Comment by alzimmermsft** [comment]
@JonathanGiles, I'll get this PR back up to speed and getting into a merge ready state.

**Comment by alzimmermsft** [comment]
Closing as this is out of date and will be worked on from a new branch

**Pull request closed without merging**